owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt

```diff
@@ -327,30 +311,56 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
         listener = null
     }
 
-    /**
-     * Get users and groups from the DB to fill in the "share with" list.
-     *
-     * Depends on the parent Activity provides a [com.owncloud.android.datamodel.FileDataStorageManager]
-     * instance ready to use. If not ready, does nothing.
-     */
-    fun refreshUsersOrGroupsListFromDB() {
-        if ((listener as BaseActivity).storageManager != null) {
-            // Get Users and Groups
-            privateShares = (listener as BaseActivity).storageManager.getPrivateSharesForAFile(
-                file?.remotePath,
-                account!!.name
-            )
+    fun updateShares(shares: ArrayList<OCShare>) {
+        updatePublicShares(
+            ArrayList(shares.filter {
+                it.shareType == ShareType.PUBLIC_LINK.value
+            })
+        )
+
+        updatePrivateShares(
+            ArrayList(shares.filter {
+                it.shareType == ShareType.USER.value ||
+                        it.shareType == ShareType.GROUP.value ||
+                        it.shareType == ShareType.FEDERATED.value
+            })
+        )
+    }
+
+    /**************************************************************************************************************
+     ************************************************ CAPABILITIES ************************************************
+     **************************************************************************************************************/
+
+    fun updateCapabilities(capabilities: OCCapability?) {
+        this.capabilities = capabilities
+
+        updatePublicLinkButton()
 
-            // Update list of users/groups
-            updateListOfUserGroups()
+        // Update view depending on updated capabilities
+        if (isPublicShareDisabled) {
```

Comment:
you can replace this if block by

`shareViaLinkSection.isGone = isPublicShareDisabled`

Reply:
Too many extensions on ktx 😮